### PR TITLE
Refactor party service

### PR DESCRIPTION
**Pull request by FabriceIRANKUNDA**
## Summary

_A summary of changes being made in this PR_

## References

closes #_insert number here_
#6290 

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] [ensure all checks pass](https://github.com/XRFoundation/XREngine/wiki/Testing-&-Contributing)
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/client-core/src/social/services/PartyService.ts [hunk 1]
```diff
@@ -10,38 +9,44 @@ import { Party } from '@xrengine/common/src/interfaces/Party'
...
 
-store.receptors.push((action: PartyActionType): any => {
-  let newValues, updateMap, partyUser, updateMapPartyUsers
-
-  state.batch((s) => {
-    switch (action.type) {
-      case 'LOADED_PARTY':
+export const PartyServiceReceptor = (action) => {
+  getState(PartyState).batch((s) => {
+    let newValues, updateMap, partyUser, updateMapPartyUsers
```
Comment (HexaField) [inline packages/client-core/src/social/services/PartyService.ts:35]: please clean up all these properties and simplify the code as much as possible

File: packages/client-core/src/systems/ui/UserMenuView.tsx [hunk 2]
```diff
@@ -103,6 +104,10 @@ const AvatarContextMenu = () => {
   const user = userState.layerUsers.find((user) => user.id.value === detailState.id.value)
   const { t } = useTranslation()
 
+  useEffect(() => {
+    addActionReceptor(PartyServiceReceptor)
```
Comment (HexaField) [inline packages/client-core/src/systems/ui/UserMenuView.tsx:108]: please add a cleanup function to this useEffect
```
return () => { 
  removeActionReceptor(PartyServiceReceptor) 
}
```

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/client-core/src/social/services/PartyService.ts [hunk 3]
```diff
@@ -10,104 +9,99 @@ import { Party } from '@xrengine/common/src/interfaces/Party'
...
         return s.merge({ party: updateMap, updateNeeded: true })
-
-      case 'PATCHED_PARTY_USER':
-        newValues = action
-        partyUser = newValues.partyUser
-        logger.info({ partyUser }, 'Patched partyUser.')
-        updateMap = _.cloneDeep(s.party.value)
+      })
+      .when(PartyAction.patchedPartyUserAction.matches, (action) => {
+        logger.info({ partyUser: action.partyUser }, 'Patched partyUser.')
```
Comment (HexaField) [inline packages/client-core/src/social/services/PartyService.ts:65]: please remove all logs like this, they are unnecessary now as all actions are automatically logged

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/client-core/src/social/services/PartyService.ts (hunk 3 above)
Comment (FabriceIRANKUNDA) [reply packages/client-core/src/social/services/PartyService.ts:65]: There are some logs from the socket, I think I can keep them.

File: packages/client-core/src/social/services/PartyService.ts [hunk 4]
```diff
@@ -10,133 +9,114 @@ import { Party } from '@xrengine/common/src/interfaces/Party'
...
 
-store.receptors.push((action: PartyActionType): any => {
-  let newValues, updateMap, partyUser, updateMapPartyUsers
-
-  state.batch((s) => {
-    switch (action.type) {
-      case 'LOADED_PARTY':
+export const PartyServiceReceptor = (action) => {
+  getState(PartyState).batch((s) => {
+    // let newValues, updateMap, partyUser, updateMapPartyUsers
```
Comment (HexaField) [inline packages/client-core/src/social/services/PartyService.ts:35]: please remove these commented out variables entirely, they are unnecessary

**Review by HexaField — CHANGES_REQUESTED** [review]

**Review by HexaField — APPROVED** [review]

**Pull request merged**